### DISCUSSION TITLE
how do we encode ghost cells such that VisIt can correctly extract them with "Inverse Ghost Zone"

### DISCUSSION THREAD

**Discussion by jfavre** [Alpine-DAV/ascent#1141] [Q&A] [answered]
I am unable to properly encode ghost zones in Ascent such that VisIt (v3.3.3) reading back the Blueprint HDF5 file would recognize the cell-data as such and show them to me with an  "Inverse Ghost Zone"

reading [VTKFormat.html](https://visit-sphinx-github-user-manual.readthedocs.io/en/develop/data_into_visit/VTKFormat.html)
it would seem like the special label "avtGhostZones" would be recognized
so I added the following to my Conduit description

`ascent_options["ghost_field_name"].append() = "avtGhostZones";

mesh["fields/avtGhostZones/association"].set("element");`

I can pseudo-color all other variables, except the "avtGhostZones" (no coloring appears). fine. But Inverse Ghost Zones does not show me any cells! 

if instead I encode my Conduit node with a more generic name, such as "ghost", i.e.
`ascent_options["ghost_field_name"].append() = "ghost";

mesh["fields/ghost/association"].set("element");`

VisIt lets me make a pseudocoloring of it (see attached screenshot for a 4 MPI task job (this is a 2D problem), but I still cannot apply 'inverse ghost zone'
![visit0000](https://user-images.githubusercontent.com/7058064/236488438-c955c7a2-3e08-463c-92d1-46c5dff54666.png)

End of the day, I would expect that Ascent would take any name I specify as "ghost_field_name" and make a blueprint HDF5 file with corresponds to what VisIt expects. I'd appreciate some guidance. TIA

**Comment by cyrush** [comment]
@jfavre  The VisIt blueprint plugin does not support ghost zone identification yet.

For ghost zones in Ascent (as you report) we do use a field to indicate ghost zones & Ascent allows you to specify this name and it will use that field as ghosts in its pipelines. 

However it does not cull ghost zones before blueprint export. 

Calling it `avtGhostZones` may actually confuse VisIt. There is some general logic in VisIt for debugging related to that name. 

I recommend giving it your own name (`ghost_zones`) and to work around the limitation in VisIt -- manually apply a threshold operator to exclude the ghosts.

**Reply by jfavre** [reply, marked as answer]
Thanks @cyrush !
